sqli.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+require_once('../_helpers/strip.php');
+
+// this database contains a table with 2 rows
+// This is my first secret (ID = 1)
+// This is my second secret (ID = 2)
+$db = new SQLite3('test.db');
+
+if (strlen($_GET['id']) < 1) {
+  echo 'Usage: ?id=1';
+} else {
+  // don't sanitize user input, making the SQL query vulnerable to
+  // an injection. The query result only returns a row count, making
+  // it blind. It can be exploited based on whether the server
+  // responds with "Yes!" or "No!"
+  $count = $db->querySingle('select count(*) from secrets where id = ' . $_GET['id']);
```

Comment:
###### [Beta]
### Security Review:
#### Vulnerability: Tainted sql string
**OWASP Categories:** A01:2017 - Injection, A03:2021 - Injection
**Confidence:** MEDIUM
**Impact:** MEDIUM
**Likelihood:** HIGH

**Description:** User data flows into this manually-constructed SQL string. User data can be safely inserted into SQL strings using prepared statements or an object-relational mapper (ORM). Manually-constructed SQL strings is a possible indicator of SQL injection, which could let an attacker steal or manipulate data from the database. Instead, use prepared statements (`$mysqli->prepare("INSERT INTO test(id, label) VALUES (?, ?)");`) or a safe library.

**Problematic Code:** 
`  $count = $db->querySingle('select count(*) from secrets where id = ' . $_GET['id']);`

**References:** CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')
**More Info & Remediation :** 
https://owasp.org/www-community/attacks/SQL_Injection

**Note :** This Vulnerability has been flagged by Semgrep. If found to be a False Positive, Please reply to this comment as `False Positive` and resolve the conversation